pkg/adapters/aiven.go

```diff
@@ -0,0 +1,542 @@
+	"max_parallel_workers":            {true, false},
+
+	// Parameters that must be applied at service level
+	"work_mem":                  {true, true},
+	"shared_buffers_percentage": {true, true}, // Instead of shared_buffers
+
+	// max_worker_processes can only be increased, not decreased
+	"max_worker_processes": {true, true},
+
+	// Known to be restricted (for documentation)
+	"max_wal_size":                 {false, false},
+	"min_wal_size":                 {false, false},
+	"random_page_cost":             {false, false},
+	"seq_page_cost":                {false, false},
+	"checkpoint_completion_target": {false, false},
+	"effective_io_concurrency":     {false, false},
+	"shared_buffers":               {false, false}, // Use shared_buffers_percentage instead
+}
+
+// CreateAivenPostgreSQLAdapter creates a new Aiven PostgreSQL adapter
+func CreateAivenPostgreSQLAdapter() (*AivenPostgreSQLAdapter, error) {
+	// Load configuration from viper
+	dbtuneConfig := viper.Sub("aiven")
+	if dbtuneConfig == nil {
+		dbtuneConfig = viper.New()
+	}
+
+	// Bind environment variables
+	dbtuneConfig.BindEnv("api_token", "DBT_AIVEN_API_TOKEN")
+	dbtuneConfig.BindEnv("project_name", "DBT_AIVEN_PROJECT_NAME")
+	dbtuneConfig.BindEnv("service_name", "DBT_AIVEN_SERVICE_NAME")
+
+	var aivenConfig adeptersinterfaces.AivenConfig
+	err := dbtuneConfig.Unmarshal(&aivenConfig)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode into struct: %v", err)
+	}
+
+	// Validate required configuration
+	err = utils.ValidateStruct(&aivenConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create Aiven client
+	aivenClient, err := aiven.NewClient(aiven.TokenOpt(aivenConfig.APIToken))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Aiven client: %v", err)
+	}
+
+	// Create default PostgreSQL adapter as base
+	defaultAdapter, err := CreateDefaultPostgreSQLAdapter()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create base PostgreSQL adapter: %v", err)
+	}
+
+	// Create adapter
+	adapter := &AivenPostgreSQLAdapter{
+		DefaultPostgreSQLAdapter: *defaultAdapter,
+		aivenConfig:              aivenConfig,
+		aivenClient:              aivenClient,
+		state: &adeptersinterfaces.AivenState{
+			LastAppliedConfig: time.Time{},
+		},
+	}
+
+	// Initialize collectors
+	adapter.MetricsState = agent.MetricsState{
+		Collectors: AivenCollectors(adapter),
+		Cache:      agent.Caches{},
+		Mutex:      &sync.Mutex{},
+	}
+
+	return adapter, nil
+}
+
+func (adapter *AivenPostgreSQLAdapter) PGDriver() *pgPool.Pool {
+	return adapter.pgDriver
+}
+
+// GetSystemInfo returns system information for the Aiven PostgreSQL service
+func (adapter *AivenPostgreSQLAdapter) GetSystemInfo() ([]utils.FlatValue, error) {
+	adapter.Logger().Info("Collecting Aiven system info")
+
+	var systemInfo []utils.FlatValue
+
+	// Get service information from Aiven API
+	ctx := context.Background()
+	service, err := adapter.aivenClient.ServiceGet(
+		ctx,
```

Comment:
As you do not re-use the context just inline it:
```suggestion
	service, err := adapter.aivenClient.ServiceGet(
		context.Background(),
```